src/Nethermind/Nethermind.Evm.Test/Eip6780Tests.cs

```diff
@@ -19,133 +19,145 @@
+
+        [SetUp]
+        public override void Setup()
         {
-            TestState.CreateAccount(TestItem.PrivateKeyA.Address, 100.Ether());
+            base.Setup();
+            TestState.CreateAccount(TestItem.PrivateKeyA.Address, 1000.Ether());
             TestState.Commit(SpecProvider.GenesisSpec);
             TestState.CommitTree(0);
-
-            Address contractAddress = ContractAddress.From(TestItem.PrivateKeyA.Address, 0);
-            byte[] initByteCode = Prepare.EvmCode
-                .ForInitOf(
-                    Prepare.EvmCode
-                        .PushData(1)
-                        .Op(Instruction.SLOAD)
-                        .PushData(1)
-                        .Op(Instruction.EQ)
-                        .PushData(17)
-                        .Op(Instruction.JUMPI)
-                        .PushData(1)
-                        .PushData(1)
-                        .Op(Instruction.SSTORE)
-                        .PushData(40)
-                        .Op(Instruction.JUMP)
-                        .Op(Instruction.JUMPDEST)
-                        .PushData(TestItem.PrivateKeyB.Address)
-                        .Op(Instruction.SELFDESTRUCT)
-                        .Op(Instruction.JUMPDEST)
-                        .Done)
+            _selfDestructCode = Prepare.EvmCode
+                .SELFDESTRUCT(TestItem.PrivateKeyB.Address)
                 .Done;
+            _contractAddress = ContractAddress.From(TestItem.PrivateKeyA.Address, 0);
+            _initCode = Prepare.EvmCode
+                .ForInitOf(_selfDestructCode)
+                .Done;
+        }
 
-            byte[] byteCode1 = Prepare.EvmCode
-                .Call(contractAddress, 100000)
-                .Op(Instruction.STOP).Done;
-
-            byte[] byteCode2 = Prepare.EvmCode
-                .Call(contractAddress, 100000)
+        [TestCase(0ul, false)]
+        [TestCase(MainnetSpecProvider.CancunBlockTimestamp, true)]
+        public void self_destruct_not_in_same_transaction(ulong timestamp, bool onlyOnSameTransaction)
+        {
+            byte[] contractCall = Prepare.EvmCode
+                .Call(_contractAddress, 100000)
                 .Op(Instruction.STOP).Done;
-
-            long gasLimit = 1000000;
-
-            EthereumEcdsa ecdsa = new(1, LimboLogs.Instance);
-            Transaction initTx = Build.A.Transaction.WithCode(initByteCode).WithValue(99.Ether()).WithGasLimit(gasLimit).SignedAndResolved(ecdsa, TestItem.PrivateKeyA).TestObject;
-            Transaction tx1 = Build.A.Transaction.WithCode(byteCode1).WithGasLimit(gasLimit).WithNonce(1).SignedAndResolved(ecdsa, TestItem.PrivateKeyA).TestObject;
-            Transaction tx2 = Build.A.Transaction.WithCode(byteCode2).WithGasLimit(gasLimit).WithNonce(2).SignedAndResolved(ecdsa, TestItem.PrivateKeyA).TestObject;
-            Block block = Build.A.Block.WithNumber(blockNumber)
+            AssertDestroyed();
```

Comment:
is this supposed to be here? or doest it check that the contract is not yet depolyed?

Reply:
correct, i left there for testing something. Removed now